overlord/snapstate/flags.go

```diff
@@ -69,13 +72,23 @@ type Flags struct {
 	// long as refresh assets are cached.
 	NoReRefresh bool `json:"no-rerefresh,omitempty"`
 
-	// RequireTypeBase is set to mark that a snap needs to be of type: base, otherwise installation fails.
+	// RequireTypeBase is set to mark that a snap needs to be of type: base,
+	// otherwise installation fails.
 	RequireTypeBase bool `json:"require-base-type,omitempty"`
+
+	// OverrideToDevModeViaModel overrides allowing a snap to be installed if it
+	// is in confinement devmode. This is set to true for currently only UC20
+	// model grades dangerous for all snaps, where we always allow devmode snaps
+	// to be installed, and installed with devmode confinement turned on.
+	// This may eventually be set for specific snaps mentioned in the model
+	// assertion for non-dangerous grade models too.
+	OverrideToDevModeViaModel bool `json:"disallow-devmode,omitempty"`
```

Comment:
the json name looks wrong?  also this is a bit of a long name? my original thinking was to call this

AllowDevMode